File: Q95_boj_5612/5612.py
n = int(input())
car = int(input())
ans = []
for i in range(n):
    plus , minus = [int(i) for i in input().split()]
    car = (car + plus - minus)
    ans.append(car)
    if ans[-1] < 0:
        ans = [0]
        break
print(max(ans))

File: Q95_boj_5612/5612.cpp
#include<iostream>

using namespace std;
int n;
int car;
int p, m;
int ans;

int main(){
	cin>>n>>car;
	for(int i =0; i< n; i++){
		cin>>p>>m;
		car = car +p - m;
		if (car < 0) {
			ans =0;
			break;
		}
		ans = (ans > car)? ans:car;
	}
	cout <<ans<<endl;
	return 0;
}
